service/frontend/nexus_handler.go

```diff
@@ -531,16 +542,43 @@ func (h *nexusHandler) nexusClientForActiveCluster(oc *operationContext, service
 	})
 }
 
-// convertNexusHandlerError converts any 5xx user handler error to a downstream error.
-func convertNexusHandlerError(t nexus.HandlerErrorType) nexus.HandlerErrorType {
-	switch t {
-	case nexus.HandlerErrorTypeDownstreamTimeout,
-		nexus.HandlerErrorTypeUnauthenticated,
-		nexus.HandlerErrorTypeUnauthorized,
-		nexus.HandlerErrorTypeBadRequest,
-		nexus.HandlerErrorTypeNotFound,
-		nexus.HandlerErrorTypeNotImplemented:
-		return t
-	}
-	return nexus.HandlerErrorTypeDownstreamError
+func handlerErrorFromClientError(err error) (string, error) {
```

Comment:
Alternatively, you could put this method on the operation context and have it set the header there.